py/server/deephaven/util/function_generated_table.py

```diff
@@ -0,0 +1,84 @@
+from typing import Callable, Union, List
+
+import jpy
+
+from deephaven import execution_context, DHError
+from deephaven.execution_context import ExecutionContext
+from deephaven.jcompat import j_lambda
+from deephaven.table import Table
+
+_J_Table = jpy.get_type("io.deephaven.engine.table.Table")
+_J_Supplier = jpy.get_type('java.util.function.Supplier')
+_J_FunctionGeneratedTableFactory = jpy.get_type("io.deephaven.engine.table.impl.util.FunctionGeneratedTableFactory")
+
+
+def create(table_generator: Callable[[], Table],
+           source_tables: Union[Table, List[Table]] = None,
+           refresh_interval_ms: int = None,
+           exec_context: ExecutionContext = None) -> Table:
+    """
+    Creates an abstract table that is generated by running the table_generator() function. The function will first be
+    run to generate the table when this method is called, then subsequently either (a) whenever one of the
+    'source_tables' ticks or (b) after refresh_interval_ms have elapsed. Either 'refresh_interval_ms' or
+    'source_tables' must be set (but not both).
+
+    The table definition must not change between invocations of the 'table_generator' function.
+
+    Note that any tables used by the 'table_generator' function *MUST* be specified in the 'source_tables'.
+
+    :param table_generator: The table generator function. This function must return a Table.
+    :param source_tables: Source tables used by the 'table_generator' function. The 'table_generator' is rerun when any of these tables tick.
+    :param refresh_interval_ms: Interval (in milliseconds) at which the 'table_generator' function is rerun.
+    :param exec_context: Optionally, a custom execution context. If 'None', the current execution context is used.
+    :return: A Table that is automatically regenerated when any of the source_tables tick. This table will be regenerated *after* any changes to the source_tables have been processed.
```

Comment:
1. Change to Google style to be consistent with the rest of the code base.
2. `source_tables` could be renamed to `trigger_tables`?
3. `exec_context` could be made more concise to `exec_ctx`; the current execution context might be absent, what happens then?
4. Would be nice to mention a typical use case

Reply:
1. done (I think)
2. It is `sourceTables` in the Java signature. I think `source` is an important distinction because any 'sources tables' used by the function must be specified here (to ensure that the engine has updated them before rerunning the function).
3. I'm not sure — whatever `with exec_context` does. Do you think I should manually throw an error here?
4. added some examples